### PR TITLE
gh-107901: Fix missing line number on BACKWARD_JUMP at the end of a for loop

### PR DESCRIPTION
**Pull request by iritkatriel**
* Issue: gh-107901

### CODE REVIEW

File: Misc/NEWS.d/next/Core and Builtins/2023-08-21-21-13-30.gh-issue-107901.hszvdk.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix missing line number on :opcode:`BACKWARD_JUMP` at the end of a for loop.
```
Comment (iritkatriel) [inline Misc/NEWS.d/next/Core and Builtins/2023-08-21-21-13-30.gh-issue-107901.hszvdk.rst]:
```suggestion
Fix missing line number on :opcode:`JUMP_BACKWARD` at the end of a for loop.
```

**Review by sweeneyde — APPROVED** [review]
LGTM!

I'm not sure how to fix the other lineless instructions you mention, but I think this is a nice simple solution for this case.

File: Python/flowgraph.c [hunk 2]
```diff
@@ -531,7 +531,7 @@ normalize_jumps_in_block(cfg_builder *g, basicblock *b) {
     if (backwards_jump == NULL) {
         return ERROR;
     }
-    basicblock_addop(backwards_jump, JUMP, target->b_label.id, NO_LOCATION);
+    basicblock_addop(backwards_jump, JUMP, target->b_label.id, last->i_loc);
```
Comment (markshannon) [inline Python/flowgraph.c:534]: This is the last physical location, not the last location executed. Won't it give the wrong line for the following?
```Py
ALWAYS_TRUE= True

def foo():
    while 1:
        if ALWAYS_TRUE:
            pass # last line executed
        else:
            pass # last physical line
```

Comment (iritkatriel) [reply Python/flowgraph.c:534]: no, `last` is the conditional backwards jump that this jump is replacing (and which is replaced by a forward jump).

**Pull request merged**